### PR TITLE
webapp plots: fix NoneType err

### LINKED ISSUE #1487: webapp plots: `TypeError: 'NoneType' object is not subscriptable` again
Seen at both these pages in internal prod:
- /benchmark-results/0650893c8a12738f8000d7b54f8e367e/
- /compare/benchmark-results/06504751231d75fb800047b403f86a19...0650893cdee7736980008597ac931f96/

```
unexpected exception, please report this:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/views.py", line 188, in dispatch_request
    return current_app.ensure_sync(meth)(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/conbench/app/_endpoint.py", line 28, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/app/conbench/app/results.py", line 283, in get
    return self.page(
           ^^^^^^^^^^
  File "/app/conbench/app/results.py", line 243, in page
    plotinfo = self.get_history_plot(
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/app/conbench/app/_plots.py", line 130, in get_history_plot
    time_series_plot(
  File "/app/conbench/app/_plots.py", line 780, in time_series_plot
    ) = get_source_for_single_benchmark_result(
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/conbench/app/_plots.py", line 920, in get_source_for_single_benchmark_result
    if not cur_run["commit"]["timestamp"]:
           ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
```

(except for the compare page, the `app/results.py` lines are replaced with corresponding lines in `app/compare.py`)

### PR DESCRIPTION
**Pull request by austin3dickey**
Fixes #1487. Also adds a few tests for these pages that I validated would have caught these errors.

### CODE REVIEW

**Comment by jgehrcke** [comment]
Thanks for adding tests!

**Review by jgehrcke — COMMENTED** [review]
:rocket:

File: conbench/app/_plots.py [hunk 1]
```diff
@@ -917,7 +917,7 @@ def get_source_for_single_benchmark_result(current_benchmark_result, cur_run, un
     # clearly on the benchmark result object we can remove the `cur_run`
     # argument and have type checking find potential bugs.
 
-    if not cur_run["commit"]["timestamp"]:
+    if not cur_run["commit"] or not cur_run["commit"]["timestamp"]:
         return None, None
```
Comment (jgehrcke) [inline conbench/app/_plots.py:921]: Right! No commit means "no timestamp", i.e. cannot position this result on the plot.

(consideration does not apply in the other way of plotting -- which we use in c-bench tinyplots for example -- where we pick the position in time using the benchmark start time). Relates to https://github.com/conbench/conbench/issues/633.

Comment (austin3dickey) [reply conbench/app/_plots.py:921]: Also relates to #1489 but now I see that's a small subset of #633!

**Pull request merged**